crates/telio-wg/src/pq.rs

```diff
@@ -0,0 +1,298 @@
+use std::{
+    collections::btree_map::Range,
+    convert::TryInto,
+    io::{self, Read, Write},
+    net::{IpAddr, Ipv4Addr, SocketAddr, SocketAddrV4},
+    ops::RangeInclusive,
+    time::Duration,
+};
+
+use boringtun::noise;
+use pnet_packet::{
+    ip::{self, IpNextHeaderProtocols},
+    ipv4::{self, Ipv4Flags, Ipv4Packet, MutableIpv4Packet},
+    udp::{self, MutableUdpPacket, UdpPacket},
+    Packet,
+};
+use pqcrypto_kyber::kyber768;
+use pqcrypto_traits::kem::{Ciphertext, PublicKey, SharedSecret};
+use rand::{prelude::Distribution, SeedableRng};
+use tokio::net::{ToSocketAddrs, UdpSocket};
+
+const SERVICE_PORT: u16 = 6480;
+const LOCAL_PORT_RANGE: RangeInclusive<u16> = 1024..=u16::MAX; // Unprivileged port range
```

Comment:
nit: maybe this should be limited to dynamic port range (49152 to 65535)?

Reply:
Sure, let me change this one

Reply:
done